dfusion_rust_core/src/models/order.rs

```diff
@@ -47,6 +47,90 @@ impl Order {
             sell_amount,
         }
     }
+
+    /// Calculates the utility of an order based on a buy and sell price as well
+    /// as the executed buy amount.
+    ///
+    /// # Returns
+    ///
+    /// Returns `Some(U256)` utility result if the inputs were valid and it
+    /// it produced a valid positive utility.
+    pub fn utility(
+        &self,
+        buy_price: u128,
+        sell_price: u128,
+        exec_buy_amount: u128,
+    ) -> Option<U256> {
+        // first lets convert everything into U256s
+        let buy_price = u128_to_u256(buy_price);
+        let sell_price = u128_to_u256(sell_price);
+        let exec_buy_amount = u128_to_u256(exec_buy_amount);
+        let buy_amount = u128_to_u256(self.buy_amount);
+        let sell_amount = u128_to_u256(self.sell_amount);
+        let fee_denominator = U256::from(2);
+
+        // adj_sell_amount =
+        //   (
+        //     (
+        //       (exec_buy_amount * buy_price) / (fee_denominator - 1)
+        //     ) *
+        //     fee_denominator
+        //   ) /
+        //   sell_price
```

Comment:
Some of the line breaks here seem excessive... are you working with a 15 character line width?

Reply:
I'll make it less line-y